docs/specs/authentication-flow-selection.md

```diff
@@ -0,0 +1,163 @@
+| --- | --- | --- | --- |
+| Defined | N/A | `x_authentication_flow_group` is present  | Selected flow group is checked against the allowlist and used if valid. |
+| Defined | N/A | `x_authentication_flow_group` is **NOT** present  | First flow group in the allowlist is used. |
+| Not defined | N/A | `x_authentication_flow_group` is present  | Request is rejected. |
+| Not defined | Defined | `x_authentication_flow_group` is **NOT** present  | First item in the each flow allowlist is used. |
+| Not defined | Not defined | `x_authentication_flow_group` is **NOT** present  | Default flow group is used. |
+
+For the latter case, the allowed flow is computed from union of flow group allowlist and flow allowlist:
+
+| Flow Group Allowlist | Flow Allowlist | Behaviour |
+| --- | --- | --- |
+| Defined | Defined | Flow group is checked against the allowlist. If valid, the flow group is used. Otherwise, the flow allowlist is used. Otherwise the request is rejected. |
+| Defined | Not defined | Flow group is checked against the allowlist. If valid, the flow group is used. Otherwise, the request is rejected. |
+| Not defined | Defined | Flow allowlist is used. |
+| Not defined | Not defined | All flows are allowed. |
+
+## Configuration
+
+### Flow Group Allowlist
+
+Each client app may have an allowlist of flow groups defined with `x_authentication_flow_group_allowlist`. Once specified, the authorization request will be rejected if specified flow group is not in the allowlist.
+
+For example, given following configuration:
+
+```yaml
+ui:
+  authentication_flow:
+    groups:
+    - name: oauth_only
+      login_flow: oauth_only
+      signup_flow: oauth_only
+      reauth_flow: password
+      promote_flow: password
+      account_recovery_flow: email
+    - name: email_password_2fa
+      login_flow: email_password_2fa
+      signup_flow: email_password_2fa
+      reauth_flow: sms_code
+      promote_flow: sms_code
+      account_recovery_flow: sms
+
+oauth:
+  clients:
+    - client_id: public_app
+      # Only allow OAuth login for public_app
+      x_authentication_flow_group_allowlist:
+      - oauth_only
+    - client_id: internal
+      # Allow both OAuth login and email_password_2fa for internal
+      x_authentication_flow_group_allowlist:
+      - oauth_only
+      - email_password_2fa
+```
+
+To authorize using a flow group, include `x_authentication_flow_group` and `x_page` in the [authentication request](/docs/specs/oidc.md#x_authentication_flow_group)
+
+In the example, client app `public_app` is only allowed to use `oauth_only` flow group:
+
+```jsonc
+{
+  // ... other fields of the authorization request
+  "x_page": "login",
+  "x_authentication_flow_group": "oauth_only"
+}
+```
+
+### Flow Allowlist
+
+Each client app may have an allowlist of flows defined with `x_authentication_flow_allowlist`. Once specified, the authorization request will be rejected if specified flow is not in the allowlist.
```

Comment:
The authorization request will not rejected, as the flow name parameter is consumed by the authflow HTTP API. Instead, the authflow HTTP API will return error, when the authflow is created with a OAuth Session of a particular client ID.

Reply:
Updated to be specific to authflow api